build-logic/src/main/kotlin/Android.kt

```diff
@@ -0,0 +1,33 @@
+import com.android.build.api.dsl.CommonExtension
+import org.gradle.api.Project
+
+fun Project.configureAndroid(
+    namespace: String,
+    androidOptions: AndroidOptions
+) {
+  plugins.apply("com.android.library")
+  extensions.findByName("android")?.apply {
+    this as CommonExtension<*, *, *, *, *>
+
+    compileSdk = getCatalogVersion("android.sdkversion.compile").toInt()
+    this.namespace = namespace
+
+    defaultConfig {
+      minSdk = if (androidOptions.withCompose) {
+        getCatalogVersion("android.sdkversion.min").toInt()
+      } else {
+        getCatalogVersion("android.sdkversion.compose.min").toInt()
```

Comment:
This looks incorrect, shouldn't the `compose.min` version be used when `androidOptions.withCompose` is true?

Reply:
100% a mistake, thanks for catching this. Fix in https://github.com/apollographql/apollo-kotlin/pull/5699